README.md

```diff
@@ -68,58 +64,200 @@ To build all examples via gradle, simply call:
-This section gives a detailed overview and links for each of these components.
+Depending on the chosen use case, only a subset of the components in this repository are used.
+
+
+## Full system architecture
+
+The full architecture includes components for multiple use cases.
+One does not need all components to realize individual use cases.
+
+[<img src="/doc/images/system-diagram-full.svg" width=80% >](https://app.diagrams.net/#Hmodelix/modelix-samples/main/doc/images/system-diagram-full.svg)
+
+In the following a short overview is given on each component.
+
+
+
+  1. **The MPS Language**
+
+     The language (meta-model) and model used in all examples.
+
+     [See the 'Language' section in the MPS README.md](mps/README.md#language)
+
+
+  2. **Generated model API (domain API)**
+
+     An API based on the meta-model generated with the [api-gen](https://github.com/modelix/api-gen) plugin from modelix.
+
+     [See the 'Generated API' section in the MPS README.md](mps/README.md#generated-api)
+
+  3. **Domain-specific OpenAPI**
+
+     In this example project, an extra domain-specific API layer is added which is defined in the [OpenAPI](https://www.openapis.org/) [specification](/openapi).
+     This layer is meant educational as no noteworthy abstractions from the language itself happen in this definition.
+     It intends to show how one introduces a clearly defined domain-specific abstraction decoupling the language engineering (meta-modeling) and the web development.
+
+     We provide two implementations of the API layer: The [`rest-api-json-bulk`](/rest-api-json-bulk) and [`rest-api-model-server`](/rest-api-model-server) components.
+
+     For more details, also see the ['OpenAPI of the Courses domain' section in the MPS README.md](openapi/README.md)
+
+
+
+  4. **OpenAPI implementation**
+
+     This project provides two implementations of the [OpenAPI](openapi) domain abstraction.
+
+     - A. MPS as a source (`rest-api-json-bulk` component)
+
+          This API implementation provides access to the model by obtaining the model knowledge directly from a running MPS instance.
+          It is implemented using ktor and connects to the [`json-bulk-model-access`](https://github.com/modelix/mps-rest-model-access) plugin running inside of MPS.
+          This component can only provide **read only access** as the `json-bulk-model-access` is read only.
+
+          For more details, also see the [`rest-api-json-bulk` README.md](rest-api-json-bulk/README.md) for details.
+
+
+
+     - B. model-server as a source (`rest-api-model-server` component)
+
+          This API implementation provides access to the model by connecting to a running [`modelix model-server`](https://github.com/modelix/modelix.core/tree/main/model-server).
+          It is implemented using Quarkus and can provide **read access** to the underlying model.
+          This is realized using websockets exposed by the `model-server`.
```

Comment:
That's not right, I would say. We have normal read access through REST and additionally a websocket for push notifications about ongoing model changes.